### DISCUSSION TITLE
Accessing Github repositories with TortoiseSVN

### DISCUSSION THREAD

**Discussion by teutonicod** [Open-Systems-Pharmacology/Forum#1039] [How-Tos]
Hi all,

Since some time I wanted to use TortoiseSVN with a repository hosted on GitHub, and I was unable to access the repository when the repository was asking for an identification. I recently found the solution and wanted to share it, in case it could be of any help to others. An identification with GitHub credentials would be asked if the repository is private or on a public repository for any commit.

When identification is requested by TortoiseSVN, it should be provided the userID for GitHub, but the password should be replaced by a personal access token (PAT). A PAT can be created following these steps

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

When creating the PAT, you will need to provide the scope of the token, which would be related to the permissions of the token itself.
For working on a public repository, you can select the option below public_repo

![image](https://user-images.githubusercontent.com/11489621/144262318-fde9af61-4cdc-484a-a24f-28cef0e1fb27.png)

For a private repository, you can select the full repository option

![image](https://user-images.githubusercontent.com/11489621/144262525-43a5fb48-68b8-46c4-bc6b-17cc532e3f5b.png)

Please notice that PAT must be saved somewhere directly after creation. Once the user closes the PAT-creation form on GitHub, the PAT cannot be shown ever again. So if you don’t write it down (or copy-paste directly), you will need to create another one. 

This identification is compatible and independent from 2-factor authentication.

Thanks a lot to @Yuri05 for clarifying some of these points! 

Hope this is helpful!

Donato

**Comment by StephanSchaller** [comment]
Thanks for sharing, @teutonicod !

**Reply by sankara-sabapathy** [reply]
Thanks!

**Comment by elmyrockers** [comment]
thanks a lot... you save my time... thanks again :)
